### ISSUE TITLE
An sql excption occured while creating layer data

### ISSUE THREAD

**Issue by kvaps** [LINBIT/linstor-server#58]
```
linstor snapshot resource restore --from-resource one-image-48 --from-snapshot one-vm-151-disk-2 --to-resource one-vm-151-disk-2
````

```
ERROR REPORT 5CE9CD22-00000-000004

============================================================

Application:                        LINBIT? LINSTOR
Module:                             Controller
Version:                            0.9.9
Build ID:                           64b60e05c91b50097963c60f88830ce504dd4fd7
Build time:                         2019-05-23T07:18:37+00:00
Error time:                         2019-05-25 23:24:38
Node:                               linstor2-controller-0
Peer:                               RestClient(127.0.0.1)

============================================================

Reported error:
===============

Category:                           Exception
Class name:                         PSQLException
Class canonical name:               org.postgresql.util.PSQLException
Generated at:                       Method 'receiveErrorResponse', Source file 'QueryExecutorImpl.java', Line #2433

Error message:                      ERROR: duplicate key value violates unique constraint "unq_ldvd_minor"
  Detail: Key (vlm_minor_nr)=(1076) already exists.

Error context:
    An sql excption occured while creating layer data

Call backtrace:

    Method                                   Native Class:Line number
    receiveErrorResponse                     N      org.postgresql.core.v3.QueryExecutorImpl:2433
    processResults                           N      org.postgresql.core.v3.QueryExecutorImpl:2178
    execute                                  N      org.postgresql.core.v3.QueryExecutorImpl:306
    executeInternal                          N      org.postgresql.jdbc.PgStatement:441
    execute                                  N      org.postgresql.jdbc.PgStatement:365
    executeWithFlags                         N      org.postgresql.jdbc.PgPreparedStatement:155
    executeUpdate                            N      org.postgresql.jdbc.PgPreparedStatement:132
    executeUpdate                            N      org.apache.commons.dbcp2.DelegatingPreparedStatement:136
    executeUpdate                            N      org.apache.commons.dbcp2.DelegatingPreparedStatement:136
    persist                                  N      com.linbit.linstor.DrbdLayerGenericDbDriver:514
    createDrbdVlmDfnData                     N      com.linbit.linstor.storage.utils.LayerDataFactory:181
    ensureDrbdRscLayerCreated                N      com.linbit.linstor.CtrlLayerStackHelper:529
    ensureStackDataExists                    N      com.linbit.linstor.CtrlLayerStackHelper:328
    create                                   N      com.linbit.linstor.VolumeDataControllerFactory:81
    createVolume                             N      com.linbit.linstor.core.apicallhandler.controller.CtrlVlmCrtApiHelper:161
    restoreOnNode                            N      com.linbit.linstor.core.apicallhandler.controller.CtrlSnapshotRestoreApiCallHandler:286
    restoreResourceInTransaction             N      com.linbit.linstor.core.apicallhandler.controller.CtrlSnapshotRestoreApiCallHandler:172
    lambda$restoreSnapshot$0                 N      com.linbit.linstor.core.apicallhandler.controller.CtrlSnapshotRestoreApiCallHandler:126
    doInScope                                N      com.linbit.linstor.core.apicallhandler.ScopeRunner:141
    lambda$fluxInScope$0                     N      com.linbit.linstor.core.apicallhandler.ScopeRunner:71
    call                                     N      reactor.core.publisher.MonoCallable:91
    trySubscribeScalarMap                    N      reactor.core.publisher.FluxFlatMap:126
    subscribe                                N      reactor.core.publisher.MonoFlatMapMany:46
    subscribe                                N      reactor.core.publisher.Flux:7734
    onNext                                   N      reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain:184
    request                                  N      reactor.core.publisher.Operators$ScalarSubscription:2041
    onSubscribe                              N      reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain:131
    subscribe                                N      reactor.core.publisher.MonoCurrentContext:35
    subscribe                                N      reactor.core.publisher.MonoFlatMapMany:49
    subscribe                                N      reactor.core.publisher.FluxOnAssembly:164
    subscribe                                N      reactor.core.publisher.FluxMapFuseable:63
    subscribe                                N      reactor.core.publisher.FluxOnErrorResume:47
    subscribe                                N      reactor.core.publisher.Flux:7734
    onNext                                   N      reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain:184
    request                                  N      reactor.core.publisher.Operators$ScalarSubscription:2041
    onSubscribe                              N      reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain:131
    subscribe                                N      reactor.core.publisher.MonoCurrentContext:35
    subscribe                                N      reactor.core.publisher.MonoFlatMapMany:49
    subscribe                                N      reactor.core.publisher.FluxContextStart:49
    subscribe                                N      reactor.core.publisher.MonoCollectList:59
    subscribe                                N      reactor.core.publisher.MonoMapFuseable:59
    subscribe                                N      reactor.core.publisher.MonoOnErrorResume:44
    subscribe                                N      reactor.core.publisher.Mono:3608
    subscribeWith                            N      reactor.core.publisher.Mono:3714
    subscribe                                N      reactor.core.publisher.Mono:3602
    subscribe                                N      reactor.core.publisher.Mono:3569
    subscribe                                N      reactor.core.publisher.Mono:3541
    doFlux                                   N      com.linbit.linstor.api.rest.v1.RequestHelper:174
    restoreResource                          N      com.linbit.linstor.api.rest.v1.SnapshotRestoreResource:65
    invoke0                                  Y      jdk.internal.reflect.NativeMethodAccessorImpl:unknown
    invoke                                   N      jdk.internal.reflect.NativeMethodAccessorImpl:62
    invoke                                   N      jdk.internal.reflect.DelegatingMethodAccessorImpl:43
    invoke                                   N      java.lang.reflect.Method:566
    lambda$static$0                          N      org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory:52
    run                                      N      org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1:124
    invoke                                   N      org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher:167
    doDispatch                               N      org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$VoidOutInvoker:159
    dispatch                                 N      org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher:79
    invoke                                   N      org.glassfish.jersey.server.model.ResourceMethodInvoker:469
    apply                                    N      org.glassfish.jersey.server.model.ResourceMethodInvoker:391
    apply                                    N      org.glassfish.jersey.server.model.ResourceMethodInvoker:80
    run                                      N      org.glassfish.jersey.server.ServerRuntime$1:253
    call                                     N      org.glassfish.jersey.internal.Errors$1:248
    call                                     N      org.glassfish.jersey.internal.Errors$1:244
    process                                  N      org.glassfish.jersey.internal.Errors:292
    process                                  N      org.glassfish.jersey.internal.Errors:274
    process                                  N      org.glassfish.jersey.internal.Errors:244
    runInScope                               N      org.glassfish.jersey.process.internal.RequestScope:265
    process                                  N      org.glassfish.jersey.server.ServerRuntime:232
    handle                                   N      org.glassfish.jersey.server.ApplicationHandler:679
    service                                  N      org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer:353
    run                                      N      org.glassfish.grizzly.http.server.HttpHandler$1:200
    doWork                                   N      org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker:569
    run                                      N      org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker:549
    run                                      N      java.lang.Thread:834


END OF ERROR REPORT.

```

**Comment by ghernadi** [comment]
Thanks for the report. I guess we have a good idea how / why this happens internally.
Until we fix this you should avoid giving a snapshot the same name as a resource.

**Comment by rp-** [comment]
This should already been fixed by e3df2b7441ee0d9bfcd43ad8a30eac71e928c4d8
